### PR TITLE
fixed revocation content-type and data format mismatch

### PR DESCRIPTION
**Pull request by malloy045**
Fixes issue #71

### CODE REVIEW

**Comment by malloy045** [comment]
@WilliamDenniss I did.  As far as I can tell, someone from the OIDF needs to sign it to be complete.

**Comment by WilliamDenniss** [comment]
OK, it should post shortly then. Thanks!

File: src/revoke_token_request.ts [hunk 1]
```diff
@@ -60,6 +61,26 @@ export class RevokeTokenRequest {
     return json;
   }
 
+
+  toStringMap(): StringMap {
```
Comment (tikurahul) [inline src/revoke_token_request.ts]: You don't need this method.

File: src/token_request_handler.ts [hunk 2]
```diff
@@ -59,7 +59,7 @@ export class BaseTokenRequestHandler implements TokenRequestHandler {
       method: 'POST',
       dataType: 'json',  // adding implicit dataType
       headers: {'Content-Type': 'application/x-www-form-urlencoded'},
```
Comment (tikurahul) [inline src/token_request_handler.ts]: Just drop the `header` from the dictionary. The implied request content type is `application/json`.

Comment (malloy045) [reply src/token_request_handler.ts]: The [OAuth2.0 Token Revocation spec](https://tools.ietf.org/html/rfc7009#section-2.1) calls for `application/x-www-form-urlencoded`.  

IdentityServer4 [enforces this](https://github.com/IdentityServer/IdentityServer4/blob/a68af9dcc0fb5b57c0ae6921670b5b56b16cb376/src/Endpoints/TokenRevocationEndpoint.cs#L69) as well.

Comment (tikurahul) [reply src/token_request_handler.ts]: If that’s the case. Drop the `dataType: json`. Let me take a look at the spec as well.

Comment (malloy045) [reply src/token_request_handler.ts]: @tikurahul the `dataType` property has been removed

File: src/token_request_handler.ts [hunk 3]
```diff
@@ -57,9 +57,8 @@ export class BaseTokenRequestHandler implements TokenRequestHandler {
     let revokeTokenResponse = this.requestor.xhr<boolean>({
       url: configuration.revocationEndpoint,
       method: 'POST',
-      dataType: 'json',  // adding implicit dataType
       headers: {'Content-Type': 'application/x-www-form-urlencoded'},
-      data: request.toJson()
+      data: this.utils.stringify(request.toStringMap())
```
[the remark this answers is not in the document]
Comment (tikurahul) [reply src/token_request_handler.ts]: That happens to be the case _only_ for the revocation request. I don’t want to change to url encoded requests for the `BaseTokenHandler`. 

The token endpoints support JSON.

[the remark this answers is not in the document]
Comment (malloy045) [reply src/token_request_handler.ts]: @tikurahul, I reverted my `dataType` changes.   I  agree with this, but your previous comment confused me:
>If that’s the case. Drop the `dataType: json`.

I also think I figured out where our communication breakdown is occurring (at least on my end) - request vs response.

- the http ***response*** should be `json`
- the ***request*** parameters should `x-www-form-urlencoded`

Are we good now?

**Comment by malloy045** [comment]
Hi, @tikurahul.  Figured I'd check back in on this.  I see merge conflicts now.  

If there's any chance that my changes will be accepted, I can handle this.  Let me know.

**Comment by tikurahul** [comment]
Sorry for the radio silence. I am on vacation with very limited connectivity. Please send me an updated PR and I will take a look when I get back. Thanks 🙏

**Comment by tikurahul** [comment]
Superseded by AppAuth 1.0.

**Pull request closed without merging**